Android/app/src/go/backend/tunnel.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"errors"
+	"io"
+	"io/fs"
+	"localhost/Intra/Android/app/src/go/intra"
+	"localhost/Intra/Android/app/src/go/intra/protect"
+	"localhost/Intra/Android/app/src/go/tuntap"
+	"log"
+	"os"
+
+	"github.com/Jigsaw-Code/outline-sdk/network"
+)
+
+// Session represents an Intra communication session.
+//
+// TODO: copy methods of intra.Tunnel here and do not expose intra package
+type Session struct {
+	// TODO: hide this internal Tunnel when finished moving everything to backend
+	*intra.Tunnel
+}
+
+func (s *Session) SetDoHServer(svr *DoHServer) { s.SetDNS(svr.tspt) }
```

Comment:
We pass the server on ConnectSession. Is this ever used?

Reply:
Yes, user can update the DoH server even during connection.

- [`GoVpnAdapter.updateDohUrl`](https://github.com/Jigsaw-Code/Intra/blob/a6e4eca3864531bc7d07ea2e974587d8f7df2d8f/Android/app/src/main/java/app/intra/net/go/GoVpnAdapter.java#L243)
- [`IntraVpnService.updateServerConnection`](https://github.com/Jigsaw-Code/Intra/blob/a6e4eca3864531bc7d07ea2e974587d8f7df2d8f/Android/app/src/main/java/app/intra/sys/IntraVpnService.java#L205)